### PR TITLE
search by name && fix some problems in events creation

### PR DESCRIPTION
**Pull request by senyoudev**
The ability to seach by event's name

### CODE REVIEW

File: NOAP/test/NOAP.js [hunk 1]
```diff
@@ -44,17 +44,19 @@ contract('NOAP', (accounts) => {
    
 
     it('creates an event', async () => {
-        const event =  await NOAPContract.connect(minterA).createEvent(tokenURINewEvent,"desc","name","morocco","meknes",true,"22/12/2022","24/12/2022","younesmeskafe@gmail.com",10);
-        console.log(event)
+        await NOAPContract.connect(minterA).createEvent(tokenURINewEvent,"desc","name","morocco","meknes",true,"22/12/2022","24/12/2022","younesmeskafe@gmail.com",10);
     });
 
     it('creates a request', async () => {
         //uint256 eventID,address attender,string date)
```
Comment (adilmezghouti) [inline NOAP/test/NOAP.js:51]: [Boulder] Make sure to remove the code that is not needed anymore.

File: src/components/Form/CreateEventForm.jsx [hunk 2]
```diff
@@ -72,11 +71,13 @@ function CreateEventForm() {
   const handleSubmit = async (e) => {
     e.preventDefault();
     setLoading(true);
+    //some validation
+    //process of uploading the image and the metadata
     try {
-      await sendFileToIPFS();
+      const url = await sendFileToIPFS();
       console.log("url",url)
```
Comment (adilmezghouti) [inline src/components/Form/CreateEventForm.jsx]: [Boulder] Make sure to remove console.log

File: src/components/Form/CreateEventForm.jsx [hunk 3]
```diff
@@ -101,14 +102,9 @@ function CreateEventForm() {
...
-    //error handling
-    if (url.trim() == "" || name.trim() == "" || description.trim() == "") {
-      NotificationManager.warning(
-        "❗Please make sure all fields are completed before creating the event."
-      );
-      return;
-    } //make metadata
+  const uploadJson = async (url) => {
+ 
+   //make metadata
```
Comment (adilmezghouti) [inline src/components/Form/CreateEventForm.jsx:73]: [Boulder] Is there a reason you removed the validation from here

File: src/components/Form/CreateEventForm.jsx [hunk 4]
```diff
@@ -164,15 +161,14 @@ function CreateEventForm() {
           url: "https://api.pinata.cloud/pinning/pinFileToIPFS",
```
Comment (adilmezghouti) [inline src/components/Form/CreateEventForm.jsx]: [Boulder] Make sure to turn this into a constant/environment variable that you can read using dotenv

**Review by adilmezghouti — APPROVED** [review]

File: src/components/Form/CreateEventForm.jsx (hunk 3 above)
Comment (senyoudev) [reply src/components/Form/CreateEventForm.jsx:73]: yes, that's why I will add a form validation in the "onSubmit function"

**Pull request merged**